### ISSUE TITLE
`indent-binary-ops`: weird conflicts with `indent`

### ISSUE THREAD

**Issue by samchouse** [eslint-stylistic/eslint-stylistic#469]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Same issue as #226

In some situations, `indent-binary-ops` and `indent` just take turns fixing each other infinitely.

The squiggly on the **correct** example shows this issue.

### Reproduction

https://eslint.style/rules/default/indent-binary-ops#examples

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by hotenov** [comment]
Sorry for off-topic _(maybe)_: 

Anybody knows how to friend `indent-binary-ops` and Prettier's the [intended behavior](https://github.com/prettier/prettier/issues/869) for multi-line **argument**?

For example, I have a code block (_formatted to satisfy Prettier_):

```js
console.log(
  "This is a long multi-line argument in Javascript which Prettier\n" +
    "which Prettier will indent with additional indent, but 'indent-binary-ops' complain on this",
);

```

Unfortunately, the `indent-binary-ops` rule wants to align with one indent _(it does not care that this is function argument 😢 )_

![image](https://github.com/user-attachments/assets/693e0353-7baf-4ac2-92a2-e5b8cafccdf3)

**Is it possibly to satisfy both** in this case (without ignoring / turning off on of them)?
